devicetypes/smartthings/philio-multiple-sound-siren.src/philio-multiple-sound-siren.groovy

```diff
@@ -0,0 +1,305 @@
+
+	def sound = settings.sound ?: state.sound
+	def duration = settings.duration ?: state.duration
+
+	if (sound != state.sound || duration != state.duration) {
+		state.sound = sound
+		state.duration = duration
+		commands << secure(zwave.configurationV1.configurationSet(parameterNumber: 31, size: 1, configurationValue: [durationMap[duration]]))
+	}
+
+	response(commands)
+}
+
+def parse(String description) {
+	log.debug "parse($description)"
+	def result = null
+	if (description.startsWith("Err")) {
+		if (zwInfo?.zw?.contains("s")) {
+			result = createEvent(descriptionText:description, displayed:false)
+		} else {
+			result = createEvent(
+					descriptionText: "This device failed to complete the network security key exchange. If you are unable to control it via SmartThings, you must remove it from your network and add it again.",
+					eventType: "ALERT",
+					name: "secureInclusion",
+					value: "failed",
+					displayed: true,
+			)
+		}
+	} else {
+		def cmd = zwave.parse(description, [0x98: 1, 0x20: 1, 0x70: 1])
+		if (cmd) {
+			result = zwaveEvent(cmd)
+		}
+	}
+	log.debug "Parse returned ${result?.inspect()}"
+	return result
+}
+
+def zwaveEvent(securityv1.SecurityMessageEncapsulation cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand([0x20: 1, 0x85: 2, 0x70: 1])
+	// log.debug "encapsulated: $encapsulatedCommand"
+	if (encapsulatedCommand) {
+		zwaveEvent(encapsulatedCommand)
+	}
+}
+
+def zwaveEvent(basicv1.BasicReport cmd) {
+	log.debug "rx $cmd"
+	handleDeviceEvent(cmd.value)
+}
+
+def zwaveEvent(notificationv3.NotificationReport cmd) {
+	def result = []
+
+	log.debug "rx $cmd"
+
+	if (cmd.notificationType == notificationv3.NotificationReport.NOTIFICATION_TYPE_BURGLAR) {
+		if (cmd.event == 3) {
+			result << createEvent(name: "tamper", value: "detected")
+		}
+		result = result + handleDeviceEvent(cmd.event)
+	} else {
+		log.warn "Unknown cmd.notificationType: ${cmd.notificationType}"
+		result << createEvent(descriptionText: cmd.toString(), isStateChange: false)
+	}
+	result
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	log.debug "Unhandled Z-Wave command $cmd"
+	createEvent(displayed: false, descriptionText: "$device.displayName: $cmd")
+}
+
+def handleDeviceEvent(value) {
+	log.debug "handleDeviceEvent $value"
+	def result = [
+		createEvent([name: "switch", value: value ? "on" : "off", displayed: false]),
+		createEvent([name: "alarm", value: value ? "both" : "off"])
+	]
+
+	// value != 0 doesn't necessarily trigger the below events,
+	// but value == 0 clears them
+	if (!value) {
+		result << createEvent([name: "chime", value: "off"])
+		result << createEvent([name: "tamper", value: "clear"])
+	}
+
+	result
+}
+
+def generateCommand(command) {
+	def sound = (command && soundMap.containsKey(command)) ? soundMap[command] : soundMap[defaultSound]
+	log.debug "Sending $command"
+	[
+		secure(zwave.notificationV3.notificationReport(notificationType: sound.notificationType, event: sound.event)),
```

Comment:
the device acts in response to a `notificationReport`?

Reply:
Yeah. It is interesting. I haven't decided if I think it is better than specific numbers sent to BasicSet, but it is kind of cool.